File: pbmx_ffi/java/io/rmf/pbmx/Share.java
package io.rmf.pbmx;

import io.rmf.pbmx.ffi.LibPbmx;
import io.rmf.pbmx.ffi.RawShare;
import com.sun.jna.Pointer;

public final class Share {
    RawShare raw;

    Share(RawShare raw) {
        this.raw = raw;
    }

    public static final class Proof {
        Proof(Pointer handle) {
            this.handle = handle;
        }

        @Override
        protected void finalize() {
            LibPbmx.INSTANCE.pbmx_delete_share_proof(this.handle);
        }

        Pointer handle;
    }
}

File: pbmx_droid/app/src/main/java/io/rmf/pbmx/Share.kt
package io.rmf.pbmx

import com.sun.jna.Pointer
import io.rmf.pbmx.ffi.FFI
import io.rmf.pbmx.ffi.RawShare

class Share internal constructor(internal var raw: RawShare) {

    class Proof internal constructor(internal var handle: Pointer) {

        protected fun finalize() {
            FFI.pbmx_delete_share_proof(this.handle)
        }
    }
}
